### ISSUE TITLE
Lint the application

### ISSUE THREAD

**Issue by zelima** [datopian/data-cli#100]
We need to follow certain styling standards. We can use XO or/and standard libs for this
* xo - https://github.com/sindresorhus/xo
* standard - https://github.com/standard/standard

## Acceptance criteria

* [x] the library's code is written according to JS standards
* [x] when I run tests, it first checks for coding standards

## Tasks

* [x] compare two above and decide which one to use
  * as we are already using `xo` in our frontend library, we decided to implement it again here
* [x] setup `xo` - install, set configs (reuse now-cli's approach)
* [x] do auto fixing + manual fixing accroding to XO report

**Comment by anuveyatsu** [comment]
FIXED in https://github.com/datahq/datahub-cli/commit/df9f750d88e76b71ea2101fe524b1412804ad865

**Comment by rufuspollock** [comment]
@anuveyatsu can you summarize a bit more when you close the issue. What was the outcome, what did you find, how is this integrated into the test stack going forward etc ...

**Comment by anuveyatsu** [comment]
Summary:

* Setup `xo` linter for the library
  * added configurations in `package.json`
  * lint runs before tests
  * added `npm run lint` script for checking lint (runs with `--quite` mode that doesn't show warnings)
* Refactoring of almost every module according to JS coding standards
  * some modules auto fixed using `xo --fix` command
  * others are fixed manually
  * in some cases skipped conventions - had setup warnings instead of errors, e.g., no camelcase for some property names that depends on server side response (post request body for aws)
